webpack.config.js

```diff
@@ -1,23 +1,51 @@
-var webpack = require('webpack');
+var webpack = require("webpack");
 
-module.exports = {
-  entry: './src/index.ts',
-  output: {
-    filename: 'index.js',
-    path: __dirname + '/vega/static',
-    library: 'nbextensions/jupyter-vega/index',
-    libraryTarget: 'amd'
+module.exports = [
+  // the main vega extension
+  {
+    entry: "./src/index.ts",
+    output: {
+      filename: "index.js",
+      path: __dirname + "/vega/static",
+      library: "nbextensions/jupyter-vega/index",
+      libraryTarget: "amd"
+    },
+    resolve: {
+      extensions: [".ts", ".tsx", ".js"]
+    },
+    devtool: "source-map",
+    module: {
+      rules: [
+        {
+          test: /\.tsx?$/,
+          loader: "ts-loader"
+        }
+      ]
+    }
   },
-  resolve: {
-    extensions: [".ts", ".tsx", ".js"]
-  },
-  devtool: 'source-map',
-  module: {
-    rules: [
-      {
-        test: /\.tsx?$/,
-        loader: 'ts-loader'
-      }
-    ]
+  // the widget extension
+  {
```

Comment:
Since this code is almost the same as above, can you extract it into a common config that gets specialized?

Reply:
Yes, I will do that.